src/makeInfinityAdder.js

```diff
@@ -3,8 +3,26 @@
 /***
  * @return {function}
  */
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(num) {
+  let sum = num || 0;
+  let sumReserve = 0;
+
+  function result(nextNum) {
+    sumReserve = 0;
+
+    if (nextNum === undefined) {
+      sumReserve = sum;
+      sum = 0;
+
+      return sumReserve;
```

Comment:
Instead of using a `sumReserve` variable, you can directly return the current `sum` and reset it. This simplifies the logic and removes the need for an additional variable.